### PR TITLE
Add auto yielding source-to-source transformation tool and add yielding support for ets:insert/2 and ets:insert_new/2

### PR DESCRIPTION
**Pull request by kjellwinblad**
This pull request consists of two commits:

* The first commit adds a new source-to-source transformation tool called Yielding C Fun (YCF) that can generate "yieldable functions"/[coroutines](https://en.wikipedia.org/wiki/Coroutine]) from functions written in a subset of C. The primary purpose of YCF is to make it easier to implement platform-independent yielding NIFs and BIFs. The file [erts/emulator/internal_doc/AutomaticYieldingOfCCode.md](https://github.com/kjellwinblad/otp/blob/fde585a283502f9a154fe090ba5d73ec28afbd34/erts/emulator/internal_doc/AutomaticYieldingOfCCode.md) describes the motivation behind YCF in more detail. YCF's documentation is located in its [README.md](https://github.com/kjellwinblad/otp/blob/fde585a283502f9a154fe090ba5d73ec28afbd34/erts/lib_src/yielding_c_fun/README.md) file.
* The second commit in this pull request utilizes YCF to add yielding support for the [ets:insert/2](http://erlang.org/doc/man/ets.html#insert-2) and [ets:insert_new/2](http://erlang.org/doc/man/ets.html#insert_new-2) functions. These functions can take a list as their second argument. The `ets:insert/2` and `ets:insert_new/2` functions did not yield even when they were given long lists, and a call to one of the functions did only consume a single reduction before commit fde585a. Commit uses YCF to add yielding support to `ets:insert/2` and `ets:insert_new/2`. The number of times a call to `ets:insert/2` or `ets:insert_new/2` yields, depends on the length of the given list and is not affected by the size of the terms in the list (this is the same type of behavior as the other ETS BIFs that support yielding have).

### CODE REVIEW

File: erts/lib_src/yielding_c_fun/main_target.mk [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+YCF_SOURCES = $(sort $(wildcard $(YCF_SOURCE_DIR)/*.c) $(YCF_EXTRA_SOURCES))
+
+YCF_OBJECTS = $(patsubst $(YCF_SOURCE_DIR)/%.c, $(YCF_SOURCE_DIR)/%.o, $(YCF_SOURCES))
+
+YCF_CFLAGS = $(filter-out -Wstrict-prototypes -Wdeclaration-after-statement -Wmissing-prototypes,$(CFLAGS))
+
+YCF_EXECUTABLE = $(YCF_SOURCE_DIR)/bin/yielding_c_fun.bin
+
+$(YCF_EXECUTABLE): $(YCF_OBJECTS)
+	$(LD) $(YCF_CFLAGS) $(LDFLAGS) $(YCF_OBJECTS) -o $@
```
Comment (garazdawi) [inline erts/lib_src/yielding_c_fun/main_target.mk]: Can we use $(V_LD) here?

File: erts/lib_src/yielding_c_fun/main_target.mk [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+$(YCF_SOURCE_DIR)/%.o: $(YCF_SOURCE_DIR)/%.c $(YCF_HEADERS)
+	$(CC) $(YCF_CFLAGS) $(LDFLAGS) $(YCF_INCLUDE_DIRS) -c $< -o $@
```
Comment (garazdawi) [inline erts/lib_src/yielding_c_fun/main_target.mk]: Can we use $(V_CC) here?

File: erts/emulator/Makefile.in [hunk 3]
```diff
@@ -625,7 +631,24 @@ $(TTF_DIR)/driver_tab.c: Makefile.in utils/make_driver_tab
 	$(gen_verbose)LANG=C $(PERL) utils/make_driver_tab -o $@ -nifs $(NIF_OBJS) $(STATIC_NIF_LIBS) -drivers $(DRV_OBJS) $(STATIC_DRIVER_LIBS)
 GENERATE += $(TTF_DIR)/driver_tab.c
 
-
+beam/erl_db_insert_list.inc: $(YCF_EXECUTABLE) beam/erl_db.c
+	$(YCF_EXECUTABLE) \
```
Comment (garazdawi) [inline erts/emulator/Makefile.in]: Can we prefix this with `$(gen_verbose)`?

File: erts/lib_src/yielding_c_fun/main_target.mk (hunk 1 above)
Comment (kjellwinblad) [reply erts/lib_src/yielding_c_fun/main_target.mk]: Yes! See commit below.

File: erts/lib_src/yielding_c_fun/main_target.mk (hunk 2 above)
Comment (kjellwinblad) [reply erts/lib_src/yielding_c_fun/main_target.mk]: Yes! See commit below.

File: erts/emulator/Makefile.in (hunk 3 above)
Comment (kjellwinblad) [reply erts/emulator/Makefile.in]: Yes! See commit below.

**Pull request merged**